js/components/user-bar.js

```diff
@@ -143,12 +143,15 @@ define([
 								};
 								this.jobListing.push(job);
 								this.jobListing.valueHasMutated();
-	
+
 							}
 						});
 					})
 					.catch(() => {
-						console.warn('The server error occurred while getting all notifications');                        
+						console.warn('The server error occurred while getting all notifications');
+					})
+					.finally(() => {
+						PollService.start(this.pollId);
```

Comment:
Why do you add `PollService.start(this.pollId);` in each place manually? Why cannot this logic (`setTimeout -> callback -> setTimeout -> ...`) be encapsulated into PollService?